website/react-magma-docs/src/pages/api/aibutton.mdx

```diff
@@ -0,0 +1,315 @@
+
+export function Example() {
+  return (
+    <ButtonGroup>
+      <AIButton leadingIcon={<SettingsIcon />}>Leading custom icon</AIButton>
+      <AIButton leadingIcon={false} trailingIcon={<SettingsIcon />}>
+        Trailing custom icon
+      </AIButton>
+      <AIButton trailingIcon={<SettingsIcon />}>Both Icons</AIButton>
+    </ButtonGroup>
+  );
+}
+```
+
+## Sizes
+
+AIButton supports `small`, `medium` (default), and `large` sizes.
+
+```tsx
+import React from 'react';
+
+import { AIButton, AIButtonSize, ButtonGroup } from 'react-magma-dom';
+
+export function Example() {
+  return (
+    <ButtonGroup>
+      <AIButton size={AIButtonSize.small}>Small</AIButton>
+      <AIButton size={AIButtonSize.medium}>Medium</AIButton>
+      <AIButton size={AIButtonSize.large}>Large</AIButton>
+    </ButtonGroup>
+  );
+}
+```
+
+## Variants
+
+There are two gradient variants: `variantA` (default) and `variantB`.
+
+```tsx
+import React from 'react';
+
+import { AIButton, AIButtonVariant, ButtonGroup } from 'react-magma-dom';
+
+export function Example() {
+  return (
+    <ButtonGroup>
+      <AIButton variant={AIButtonVariant.variantA}>Variant A</AIButton>
+      <AIButton variant={AIButtonVariant.variantB}>Variant B</AIButton>
+    </ButtonGroup>
+  );
+}
+```
+
+## Shapes
+
+Shapes include `fill` (default), `leftCap`, `rightCap`, and `round` (used for icon-only buttons).
+
+```tsx
+import React from 'react';
+
+import { AIButton, AIButtonShape, ButtonGroup } from 'react-magma-dom';
+
+export function Example() {
+  return (
+    <ButtonGroup>
+      <AIButton shape={AIButtonShape.fill}>Fill</AIButton>
+      <AIButton shape={AIButtonShape.leftCap}>Left Cap</AIButton>
+      <AIButton shape={AIButtonShape.rightCap}>Right Cap</AIButton>
+      <AIButton shape={AIButtonShape.round} leadingIcon aria-label="Round" />
+    </ButtonGroup>
+  );
+}
+```
+
+## Text Transform
+
+Text can be `uppercase` (default) or `none`.
+
+```tsx
+import React from 'react';
+
+import { AIButton, AIButtonTextTransform, ButtonGroup } from 'react-magma-dom';
+
+export function Example() {
+  return (
+    <ButtonGroup>
+      <AIButton textTransform={AIButtonTextTransform.uppercase}>
+        UPPERCASE
+      </AIButton>
+      <AIButton textTransform={AIButtonTextTransform.none}>
+        Normal Text
+      </AIButton>
+    </ButtonGroup>
+  );
+}
+```
+
+## Full Width
+
+Set `isFullWidth` to make the button expand to the container width.
+
+```tsx
+import React from 'react';
+
+import { AIButton } from 'react-magma-dom';
+
+export function Example() {
+  return <AIButton isFullWidth>Full Width</AIButton>;
+}
+```
+
+## Animated Gradient
+
+The `isAnimated` prop can be used to set gradient animation to background in the AI Button.
+Because of the problems that animation can create for people who are sensitive to such things,
+we automatically disable it in the AI Button if we detect that the `prefers-reduced-motion` media query is set.
```

Comment:
Maybe we can summarize this with `We automatically turn off the animation for users who have Reduced Motion turned on.`